processing/src/main/java/io/druid/segment/DimensionMerger.java

```diff
@@ -0,0 +1,137 @@
+   *
+   * @param adapters List of adapters to be merged.
+   * @throws IOException
+   */
+  public void mergeValueMetadataAcrossSegments(List<IndexableAdapter> adapters) throws IOException;
+
+
+  /**
+   * Convert a row from a single segment to its equivalent representation in the merged set of rows.
+   *
+   * This function is used by the index merging process to build the merged sequence of rows.
+   *
+   * The implementing class is expected to use the merged value metadata constructed
+   * during mergeValueMetadataAcrossSegments, if applicable.
+   *
+   * For example, an implementation of this function for a dictionary-encoded String column would convert the
+   * segment-specific dictionary values within the row to the common merged dictionary values
+   * determined during mergeValueMetadataAcrossSegments().
+   *
+   * @param segmentRow A row from a segment to be converted to its representation within the merged sequence of rows.
+   * @param segmentIndexNumber Integer indicating which segment the row originated from.
+   */
+  public Object convertSegmentRowValuesToMergedRowValues(Object segmentRow, int segmentIndexNumber);
+
+
+  /**
+   * Process a row from the merged sequence of rows and update the DimensionMerger's internal state.
+   *
+   * After constructing a merged sequence of rows across segments, the index merging process will
+   * iterate through these rows and pass row values from each dimension to their correspodning DimensionMergers.
+   *
+   * This allows each DimensionMerger to build its internal view of the sequence of merged rows, to be
+   * written out to a segment later.
+   *
+   * @param rowValues The row values to be added.
+   * @throws IOException
+   */
+  public void processMergedRow(Object rowValues) throws IOException;
+
+
+  /**
+   * Internally construct any index structures relevant to this DimensionMerger.
+   *
+   * After receiving the sequence of merged rows via iterated processMergedRow() calls, the DimensionMerger
+   * can now build any index structures it needs.
+   *
+   * For example, a dictionary encoded String implementation would create its bitmap indexes
+   * for the merged segment during this step.
+   *
+   * The index merger will provide a list of row number conversion IntBuffer objects.
+   * Each IntBuffer is associated with one of the segments being merged; the position of the IntBuffer in the list
+   * corresponds to the position of segment adapters within the input list of mergeValueMetadataAcrossSegments().
+   *
+   * For example, suppose there are two segments A and B.
+   * Row 24 from segment A maps to row 99 in the merged sequence of rows,
+   * The IntBuffer for segment A would have a mapping of 24 -> 99.
+   *
+   * @param segmentRowNumConversions A list of row number conversion IntBuffer objects.
+   *
+   * @throws IOException
+   */
+  public void buildIndexes(List<IntBuffer> segmentRowNumConversions) throws IOException;
```

Comment:
This writes indexes, right? If so let's put "write" in the name, like `writeIndexes`